CMakeLists.txt

```diff
@@ -64,6 +78,16 @@ set(QUIC_BUILD_PERF "OFF")
 
 set(QUIC_TLS_SECRETS_SUPPORT "ON")
 
+configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/quicer_vsn.hrl.in
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/quicer_vsn.hrl
```

Comment:
yeah, it is confusing. 
CMAKE_BINARY_SOURCE_DIR won't work for rebar3 as it is only understood by cmake while building C objects. In other words, rebar3 won't read from CMAKE_BINARY_SOURCE_DIR.

We may let rebar3 export the build path in use and use it here or add an include path in compile opts but I just think it is over complicated since the file is just one liner.